### PR TITLE
Support process.runtime.* resource attributes

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Resolves https://github.com/open-telemetry/opentelemetry-java/issues/1908

### CODE REVIEW

File: sdk-extensions/resources/src/main/java/io/opentelemetry/sdk/extension/resources/ProcessRuntimeResource.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+/** {@link ResourceProvider} which provides information about the Java runtime. */
+public final class ProcessRuntimeResource extends ResourceProvider {
+  @Override
+  protected Attributes getAttributes() {
+    try {
+      String name = System.getProperty("java.runtime.name");
+      String version = System.getProperty("java.runtime.version");
+      String description =
+          String.join(
```
Comment (Oberon00) [inline sdk-extensions/resources/src/main/java/io/opentelemetry/sdk/extension/resources/ProcessRuntimeResource.java:23]: I think I would slightly prefer using `+` here, but no strong opinion.

File: sdk-extensions/resources/src/test/java/io/opentelemetry/sdk/extension/resources/ProcessRuntimeResourceTest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import org.junit.jupiter.api.Test;
+
+class ProcessRuntimeResourceTest {
+  @Test
+  void shouldCreateRuntimeAttributes() {
+    // when
+    ReadableAttributes attributes = new ProcessRuntimeResource().getAttributes();
+
+    // then
+    assertThat(attributes.get(ResourceAttributes.PROCESS_RUNTIME_NAME)).isNotNull();
```
Comment (Oberon00) [inline sdk-extensions/resources/src/test/java/io/opentelemetry/sdk/extension/resources/ProcessRuntimeResourceTest.java:22]:
```suggestion
    assertThat(attributes.get(ResourceAttributes.PROCESS_RUNTIME_NAME)).isNotBlank();
```

Slightly stricter assertion: https://joel-costigliola.github.io/assertj/core/api/org/assertj/core/api/AbstractCharSequenceAssert.html#isNotBlank()

**Review by Oberon00 — APPROVED** [review]
I think that's nice: We have both a ResourceProvider and it is included in the default attributes.
The only problem I can see over "hard-coding" this like the telemetry SDK resource, is that the SPI approach might not work with some fat Jar approaches.

File: sdk-extensions/resources/src/test/java/io/opentelemetry/sdk/extension/resources/ProcessRuntimeResourceTest.java (hunk 2 above)
Comment (mateuszrzeszutek) [reply sdk-extensions/resources/src/test/java/io/opentelemetry/sdk/extension/resources/ProcessRuntimeResourceTest.java]: Good point, done.

File: sdk-extensions/resources/src/main/java/io/opentelemetry/sdk/extension/resources/ProcessRuntimeResource.java (hunk 1 above)
Comment (mateuszrzeszutek) [reply sdk-extensions/resources/src/main/java/io/opentelemetry/sdk/extension/resources/ProcessRuntimeResource.java]: I'm fine with both versions.

**Review by anuraaga — APPROVED** [review]
Nice

**Review by jkwatson — APPROVED** [review]
Thanks!

**Pull request merged**